src/guitimer_func.h

```diff
@@ -0,0 +1,61 @@
+/* $Id$ */
+
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file guitimer_func.h GUI Timers. */
+
+#ifndef GUITIMER_FUNC_H
+#define GUITIMER_FUNC_H
+
+class GUITimer
```

Comment:
Really needs some usage documentation. "Multipurpose timer for GUI functions, mainly for use with `Window::OnRealtimeTick` events. Should either use only `CountElapsed` or `Elapsed` to signal that time has passed. If `CountElapsed` is used, the timer functions as a multi-shot repeater, while using `Elapsed` has the trigger function as single-shot.